### ISSUE TITLE
structure of the run, rezzing just before access

### ISSUE THREAD

**Issue by prozz** [mtgred/netrunner#334]
there is a need of new step when run is almost finished:
- runner should be asked: 'Access?' with answers 'Yes' or 'Jackout'
- when 'Yes' corp got window for rezzing DRT or Crisium Grid and confirms successful run.

right now there is a confusion, after passing all ices when i rez DRT runner still sees 'Jackout' option and this leads to misunderstandings.

if this issue can be sorted with current implementation, pls let me know. im happy to gather such things and write a faq for the game ;)

regards
PR

**Comment by JoelCFC25** [comment]
Related to #131.

**Comment by mtgred** [comment]
I prefer to avoid adding an additional step as it is rarely needed and would make all runs more tedious.

A solution would be to ask the Runner in the chat when he passes the last ICE if he want to access before clicking "No more action". It could look suspicious, but it could be a bluff as well. It's basically how you would do it in real life.

Another solution would be to disable the jack out button when the Runner decides to go pass the last ICE. But I prefer leave jack out button always available to allow the Runner to cancel the run in case something goes wrong.

**Comment by prozz** [comment]
hmm. valid reasoning.

got another idea then, that comes along with 'thinking button' suggested in another issue.

how about a set of very common phrases that are built-in as keyboard shotcuts or even some cute buttons? examples would include:
- thinking
- access?
- afk/brb

id love to be able to use more keyboard for both that and some common operations like drawing or taking credits.

**Comment by thehunter518** [comment]
I was going to say that asking the runner "do you want to access?" makes them immediately suspicious and spoils the trap, but mtgred makes a good point - that is literally how it works face to face. In real life I usually try to say something innocuous like "alright, so you're in?". I've never really considered doing it to full-on bluff someone, but it's a valid option.

I think if it was implemented on the site as an additional button prompt, the advantage would go to the corp, as the runner would quickly grow weary of pressing "Continue" multiples times during a run, and wouldn't ever think to jack out. I also see how it would make every single run in the game more tedious for the sake of one or two cards. 

I'm just going to start asking every time I guess, to get in their head. Whenever my friend plays Wayland (in-person), he constantly asks the runner how many cards they have in their hand. Same idea I think, just a bit more tedious to type it out.

**Comment by aiannacc** [comment]
The way it works in person is **wrong**.  The runner should be required to declare his intentions at the moment he passes the last ice or runs on an unprotected server.  He should not be able to bait the corp into rezzing a Dedicated Response Team before jacking out, whether through inadvertence or devious intent.

The "are you accessing?" question so often asked in person is a necessary compromise of live play, but it's easy to avoid in an online version.  In fact, the necessary dialog already exists in the current implementation.  Runs need not be made more tedious at all... that dialog just needs to be modified.

Just replace the dialog that the runner faces after he passes the last ice.  At present it says "Continue/Jack Out."  It should instead say "Access/Jack Out."  For runs on unprotected servers, just add a "Jack out before access?" check box.

Finally, note that keeping the "Jack Out" button prominently in front of the runner introduces terrible confusion for new players.  From the current UI, novices naturally conclude that they are allowed to postpone their access decision until after the corp has rezzed any assets or declared "No More Action."  This actually *encourages* novice runners to think of baiting the corp with illegal feints as a legitimate strategic aspect of the game.

**Comment by zaroth** [comment]
I can't agree with @mtgred it's a rarely needed window - unless you're trying to say that Jackson Howard is rarely used or trapping archives is uncommon. Just yesterday I had a runner jack out from accessing my (visibly) double-Shocked, double-ShiKyu'd archives after I used Howard to shuffle in (facedown) archived cards. One could argue that corp NEVER wants to use Howard's ability in 4.1 window in this situation - and if the Howard was unrezzed, the corp wasn't even able to do it during 4.1 window, as opposed to 4.3.

I agree with @mtgred too that we need an emergency jackout "button" at all times (especially with the current state of cards implementation). However, it doesn't need to be a prominent part of the UI - as @aiannacc said, having a huge button with "Jack out" 'legitimizes' the illegal action, especially for newbies. A good alternative for an 'emergency jackout' would be a console command, following the example of the recently introduced, e.g. **/endrun** - ends the current run and writes to chat window something along the lines "Corp/runner uses a debug command to end the current run." On a bit related note, the console commands could be made a bit more accessible via adding **/help** command and/or a visual "debug" menu.

As for real life play relevance, true, I already got quite used to asking short "You sure?" before every access, probably thanks to the type of corp decks I enjoy playing. However, annoyance of saying this IRL vs typing on keyboard in a primarily mouse operated computer game is incomparable.

So, in short, I'm proposing the following:

1. use the @aiannacc suggestion of replacing existing dialogs to prevent adding additional tediousness
2. instead, adding another emergency-kind-of-way to end the run in windows where it's not normally allowed, such as **/endrun**, not as prominent, as not to teach bad habits and confuse new players.

**Comment by vildninja** [comment]
The way I see it, leaving out step 4.1 in a run is a very relevant error. The issue is more related to #131 since account siphon and many other event's do not access cards, so asking if the player want to access cards can be misleading. (I would say no if I ran with Eater, and I would still expect a successful run)

Yesterday I ran into the runner accessing TGTBT from RnD, so allowing delayed rezzing would have allowed me to cheat, if the rez window was prolonged.

I know it is an annoying step to add everywhere, but 4.1 is a very relevant step in any trap based run.

**Comment by vildninja** [comment]
Just gave it an extra thought last night, and I think the best solution would be to add an extra button on the corp side, along with the "no further action" button. So instead the corp has two buttons, one just saying "no further action" as there is now, and one saying "ask again when successful".

Not too happy with the wordings though. But this would allow the corp to open a 4.1 window only when needed, while still maintaining the smooth flow of the current system for the remaining 95% of all runs (yes I just made that number up)

i.e.
1)
corp
[no further actions] - goto 4
[ask when successful] - goto 2

2)
runner (if corp requested 4.1 window)
[successful] - goto 3
[jack out] - jack out

3)
runner show buttons greyed out
corp this is window 4.1
[no further actions] - goto 5

4)
runner (if corp ignored 4.1 window)
[successful] - goto 5
[jack out] - jack out

5)
runner show buttons enabled
[access cards]
[card abilities]

@mtgred I did ask the runner in chat yesterday, and got him confused because he couldn't press the successful button before I pressed no further actions. But yes talking more in the chat is also a valid option.

**Comment by muyjohno** [comment]
+1 for requesting a corp ability opportunity between "successful run" being confirmed and access. It's far from rare to use it.

**Comment by Agnate** [comment]
+1 to @vildninja idea of asking the Corp if they want the extra step or not. It satisfies @mtgred argument about not adding unnecessary actions for the Runner and addresses the issue of not having the 4.1 step. You lose a bit of that bluffing aspect, but having to type less into chat is better for a game like Netrunner.

**Comment by nealterrell** [comment]
I have cooked up a solution to this as part of my start-of-turn revamp in #282. It works essentially how  @vildninja described above: when the runner hits position 0 (past last ice), Corp gets buttons for "No Action" and for "Action before access". If the click the latter, the runner (who doesn't know which option was selected) will see a "Waiting for Corp" prompt upon selecting Successful Run. The access routine will be delayed until:

* the Corp takes his actions, clicking Done when finished. (Example: rez Dedicated Response Team.)
* the Runner then takes actions, clicking "Continue" when finished. There is no "Jack Out" button in this step. (Example: pay for Plascrete Carapace off Personal Workshop.)
* after the last Continue, the normal access prompts kick in.

No extra steps are required if the Corp doesn't want to act in 4.3. If they do want to act, then it's +1 click to each player.

**Comment by nealterrell** [comment]
Submitted in #1218.
